plugin/ipam/main.go

```diff
@@ -110,7 +113,7 @@ func cmdAdd(args *skel.CmdArgs) error {
 	if alloc == nil {
 		// allocate IPs on an available interface
 		allocs, err := aws.DefaultClient.AllocateIPsFirstAvailableAtIndex(conf.IfaceIndex, conf.IPBatchSize)
-		if err == nil {
+		if err == nil || len(allocs) > 0 {
```

Comment:
why not &&?

Reply:
I updated `AllocateIPsFirstAvailableAtIndex` to return partial results, the only place where there's an alloc and also an error is when we successfully allocate an IP but fail to track it in the registry, we were dropping the registry error before, and now that we're returning it if we got an error but still have an IP we should just use that IP, otherwise we'd create a new interface.